dask/base.py

```diff
@@ -1359,19 +1397,7 @@ def normalize_array(x):
                     # bytes fast-path
                     data = hash_buffer_hex(b"-".join(x.flat))
             except (TypeError, UnicodeDecodeError):
-                try:
-                    data = hash_buffer_hex(pickle.dumps(x, pickle.HIGHEST_PROTOCOL))
-                except Exception:
-                    # pickling not supported, use UUID4-based fallback
-                    if not config.get("tokenize.ensure-deterministic"):
-                        data = uuid.uuid4().hex
-                    else:
-                        raise RuntimeError(
-                            f"``np.ndarray`` with object ``dtype`` {x!r} cannot "
-                            "be deterministically hashed. Please, see "
-                            "https://docs.dask.org/en/latest/custom-collections.html#implementing-deterministic-hashing "  # noqa: E501
-                            "for more information"
```

Comment:
This feels like an unrelated change. While the functionality is similar, the raised exception is a little different

Reply:
Blocked by and incorporates https://github.com/dask/dask/pull/10883.
To review this PR before the above is merged, just look at the last commit.

Reply:
This PR incorporates https://github.com/dask/dask/pull/10883.
To review this PR before the above is merged, just look at the last commit.
This comment is about a change in #10883; please keep the critique there.